leshan-server-core/src/test/java/org/eclipse/leshan/server/queue/PresenceServiceTest.java

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ *
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ *
+ * Contributors:
+ *     Bosch Software Innovations GmbH - initial API
+ *******************************************************************************/
+package org.eclipse.leshan.server.queue;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.leshan.core.request.BindingMode;
+import org.eclipse.leshan.server.registration.Registration;
+import org.junit.Test;
+
+/**
+ * tests the implementation of {@link PresenceService}
+ *
+ */
+public class PresenceServiceTest {
+
+    private PresenceService instance = new PresenceServiceImpl();
+
+    @Test
+    public void testSetOnlineForNonQueueMode() throws Exception {
+        Registration registration = givenASimpleClient();
+        final CountDownLatch countDown = new CountDownLatch(2);
+        instance.addListener(new PresenceListener() {
+
+            @Override
+            public void onOnline(Registration endpoint) {
+                countDown.countDown();
+            }
+
+            @Override
+            public void onOffline(Registration endpoint) {
+                countDown.countDown();
+            }
+        });
+        instance.setOnline(registration);
+        assertFalse(countDown.await(2, TimeUnit.SECONDS));
+    }
+
+    /**
+     * Test method for
+     * {@link org.eclipse.leshan.server.queue.PresenceService#setOffline(org.eclipse.leshan.server.client.Registration)}.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testSetOnlineAndOffline() throws Exception {
+        Registration queueModeRegistration = givenASimpleClientWithQueueMode();
+        final CountDownLatch countDown = new CountDownLatch(2);

```

Comment:
I suppose a counter for each state should be ok.
You can also call several time setOffline or setOnline and see that the counter is not incremented.